src/codeExecutionWorker.js

```diff
@@ -0,0 +1,67 @@
+/* global eval */
+'use strict'
+import { connectIframe, PluginClient } from '@remixproject/plugin'
+import ethers from 'ethers' // eslint-disable-line
+import Web3 from 'web3' // eslint-disable-line
+import swarmgw_fn from 'swarmgw' // eslint-disable-line
+window.swarmgw = swarmgw_fn() // eslint-disable-line
+window.ethers = ethers
+window.Web3 = Web3
+
+class Web3Provider {
+  constructor (plugin) {
+    this.plugin = plugin
+  }
+  sendAsync (payload, callback) {
+    this.plugin.call('appWeb3Provider', 'sendAsync', payload).then(result => callback(null, result)).catch(e => callback(e.message))
+  }
+}
+
+class CodeExecutor extends PluginClient {
+  execute (script) {
+    if (!this.currentRequest.isFromNative) return
```

Comment:
`isFromNative` will be remove in next version. Consider your plugin as an open API that other plugins should be able to use. Else don't create a plugin.
Let the developer decide if s.he want the plugin to be called.

Reply:
`isFromNative` is going to be replaced by something else?

Reply:
cause I remember we have this for a specific reason